src/main/java/org/passay/NumberRangeRule.java

```diff
@@ -0,0 +1,135 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.passay;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Rule for determining if a password contains any number within a defined range, lower inclusive, upper exclusive.
+ *
+ * @author  Middleware Services
+ */
+public class NumberRangeRule implements Rule
```

Comment:
Ah, I didn't recall that use case during my review. I don't think the current implementation supports that use case, though. In any case that probably justifies this rule. I still think `IllegalSequenceRule` could benefit from `StringMatch`, though possibly in a separate PR.